### DISCUSSION TITLE
[Question] webpack splitchunks import node_modules

### DISCUSSION THREAD

**Discussion by hasban12138** [webpack/webpack#14463] [Q&A] [answered]
`webpack: 4.35.3`
I have a folder one and some files import modules. now I use splitChunks.cacheGroups.test like this
```ts
test: (module) => {
    return (
       /src[\\/]one[\\/]/i.test(module.context)
    );
},
```
 if I use test I can't bundle modules in one.

It has any way fix it?

Thanks!

**Comment by alexander-akait** [comment, marked as answer]
Sorry, hard to understand, can you clarify?

**Reply by hasban12138** [reply]
sorry
I mean I use splitChunks and I want to Split file
e.g.
```ts
+- one
|---one.tsx

+-two
|---two.tsx
```
and in webpack like
```ts
entry: {
  one:one.tsx
  two:two.tsx
}

optimization: {
  splitChunks: {
    cacheGroups: {
      one: {
          name：‘one’,
          test: (module) => {
              return (
                 /src[\\/]one[\\/]/i.test(module.context)
              );
          },
       }
     }
  }
}
```
but it have a problem , if I use test , File one.tsx can't bundle node_module in one.bundle.js and two.bundle.js alse don't have one.tsx
so it has any way get it ？
Thanks！

**Reply by alexander-akait** [reply]
Because it is initial entry point, what you try to achieve?

**Reply by hasban12138** [reply]
I want one.tsx as a base file , and I can use it by itself  or  in three.tsx even four.tsx use it
Is there any other ways to make it ?

**Reply by alexander-akait** [reply]
What is `three.tsx`/`four.tsx`? Do you understand how splitting is working?

**Reply by hasban12138** [reply]
I mean `three.tsx/four.tsx` is `three.bundle.js/four.bundle.js` they have `one.tsx`  , I can just need to use `three.bundle.js` to get `one.tsx`  and in `one.bundle.js` will not be affected

**Reply by alexander-akait** [reply]
chunks: 'all'?
